pkg/db/sql_api.go

```diff
@@ -0,0 +1,63 @@
+package canD
+
+import (
+	"database/sql"
+	"telemetry/pkg/msgs"
+	"time"
+)
+
+// WriteMsg commits a CAN message to the database.
+func WriteMsg(db *sql.DB, msg msgs.CAN) error {
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+
+	insert, err := tx.Prepare("INSERT INTO can(id, timestamp, data) VALUES(?, ?, ?)")
+	if err != nil {
+		return err
+	}
+
+	defer insert.Close()
+
+	insert.Exec(msg.ID, msg.Timestamp, msg.Data)
+
+	tx.Commit()
+	return nil
+}
+
+// TimeWindowedRead reads msgs between to and from with the provided id.
+func TimeWindowedRead(db *sql.DB, canID uint16, to time.Time, from time.Time) ([]msgs.CAN, error) {
```

Comment:
It feels kind of weird to have `to` before `from`. I think the convention is usually `(start, end)` when providing a windowed specification that defines partitioning/ordering/frame boundaries?